Mathlib/Data/Set/ToFunction.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2024 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Data.Rel
+import Mathlib.Logic.ExistsUnique
+
+/-!
+# Set to function
+
+This file provides API for sets that define functions.
+
+## Main definitions
+
+- `Set.asPartialFun` converts `Set (α × β)` to `α → Option β` if possible.
+- `Set.asFun` converts `Set (α × β)` to `α → β` if possible.
+
+These definitions mimic the standard definition of functions in set theory.
+
+## Main theorems
+
+- `Function.graph_toSet_asFun` says that if you convert a function to a graph, the graph to a set,
+and the set to a function, you will obtain the same function.
+-/
+
+variable {α β : Type*}
+
+namespace Set
+
+section set_as_partial_function
+
+/-- A set `s : Set (α × α)` represents a partial function if for every `x : α` there is at most one
+`y : β` such that `(x, y) ∈ s`. -/
+def IsPartialFun (s : Set (α × β)) : Prop :=
+  ∀ x : α, { y : β | (x, y) ∈ s }.Subsingleton
+
+open Classical in
+/-- Use given set on `α × α` as a partial function. Each `x : α` is mapped to the unique `y : β`
+such that `(x, y) ∈ s`, or to `none` if none exists.  -/
+noncomputable def asPartialFun (s : Set (α × β)) : α → Option β :=
+  fun a : α => if hb : ∃ b, (a, b) ∈ s then hb.choose else none
```

Comment:
Did you consider `PFun` here?

Reply:
You mean as a name for the noncomputable definition?